impl/doc/try_from.md

```diff
@@ -0,0 +1,36 @@
+# What `#[derive(TryFrom)]` generates
+
+This derive allows you to convert enum discriminants into their corresponding variants.
+A requirement is, that the enum has a `#[repr(u/i*)]` attribute, e.g., `#[repr(u8)]`.
+Only field-less variants can be constructed from their variant, therefor the `TryFrom` implementation will return an error for a discriminant representing a variant with fields.
+
+## Example usage
+
+```rust
+# #[rustversion::since(1.66)]
+# mod discriminant_on_non_unit_enum {
+# use derive_more::TryFrom;
+#[derive(TryFrom, Debug, PartialEq)]
+#[repr(u32)]
+enum Enum {
+    Implicit,
+    Explicit = 5,
+    Field(usize),
+    Empty{},
+}
+
+# #[rustversion::since(1.66)]
+# pub fn test(){
+assert_eq!(Enum::Implicit, Enum::try_from(0).unwrap());
+assert_eq!(Enum::Explicit, Enum::try_from(5).unwrap());
+assert_eq!(Enum::Empty{}, Enum::try_from(7).unwrap());
+
+// variants with fields are not supported
+assert!(Enum::try_from(6).is_err());
+# }
+# }
+# // We need to use a `function` declaration, because we cannot put `rustversion` on a statement.
+# #[rustversion::since(1.66)] use discriminant_on_non_unit_enum::test;
+# #[rustversion::before(1.66)] fn test() {}
+# test();
```

Comment:
@JelteF I'm really not sure how to do this better, the problem is, that 1.65 doesn't allow discriminants on enums with fields.

Reply:
@ModProg @JelteF I think it's worth bumping MSRV up to 1.66.

Reply:
On the one hand it's only documentation, on the other hand, it's only one version.

Reply:
I would say let's just bump it to 1.72, by the time we release 1.0 I think that will have been out long enough.